endpointmanager/pkg/capabilityhandler/capabilityreceiver.go

```diff
@@ -141,15 +142,21 @@ func saveMsgInDB(message []byte, args *map[string]interface{}) error {
 		return fmt.Errorf("unable to cast context from arguments")
 	}
 
-	existingEndpt, err = store.GetFHIREndpointUsingURL(ctx, fhirEndpoint.URL)
+	endptSrc, err = store.GetFHIREndpointUsingURL(ctx, url)
+	if err != nil {
+		return errors.Wrapf(err, "unable to find fhir endpoint with url %s", url)
+	}
+
+	fhirEndpoint.FHIREndpointID = endptSrc.ID
+	existingEndpt, err = store.GetFHIREndpointInfoUsingFHIREndpointID(ctx, endptSrc.ID)
 
 	// If the URL doesn't exist, add it to the DB
```

Comment:
I personally find this comment confusing now since it errors out if the URL doesn't exist, but this feels very minor and is my only comment so if you just want to resolve this without changing anything i'm cool with that too!